src/Composer/Package/Loader/RootPackageLoader.php

```diff
@@ -93,6 +99,14 @@ public function load(array $config, string $class = 'Composer\Package\RootPackag
             }
 
             if (!isset($config['version'])) {
+                if ($this->io !== null && $config['name'] !== '__root__') {
+                    $this->io->warning(
+                        sprintf(
+                            "Composer couldn't find the root package version for %s, defaulting to `1.0.0`. See https://getcomposer.org/doc/articles/troubleshooting.md#dependencies-on-the-root-package for more details.",
```

Comment:
Let's use a shortcut here, I'll make sure it works later.

```suggestion
                            "Composer couldn't find the root package version for %s, defaulting to `1.0.0`. See https://getcomposer.org/root-version",
```